src/components/numeral-date/numeral-date.test.tsx

```diff
@@ -0,0 +1,1995 @@
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "", mm: "", yyyy: "" }}
+          onChange={() => {}}
+          enableInternalWarning
+        />
+      </CarbonProvider>
+    );
+
+    const dayInput = screen.getByRole("textbox", { name: "Day" });
+    const yearInput = screen.getByRole("textbox", { name: "Year" });
+    await user.type(dayInput, "32");
+    await user.type(yearInput, "2201");
+    await user.tab();
+
+    expect(
+      screen.getByText(
+        "Day should be a number within a 1-31 range. " +
+          "Year should be a number within a 1800-2200 range."
+      )
+    ).toBeVisible();
+  });
+
+  it("should render the appropriate validation message when the `Month` and `Year` fields are blurred and have values greater than the valid limits", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "", mm: "", yyyy: "" }}
+          onChange={() => {}}
+          enableInternalWarning
+        />
+      </CarbonProvider>
+    );
+
+    const monthInput = screen.getByRole("textbox", { name: "Month" });
+    const yearInput = screen.getByRole("textbox", { name: "Year" });
+    await user.type(monthInput, "13");
+    await user.type(yearInput, "2201");
+    await user.tab();
+
+    expect(
+      screen.getByText(
+        "Month should be a number within a 1-12 range. " +
+          "Year should be a number within a 1800-2200 range."
+      )
+    ).toBeVisible();
+  });
+
+  it("should render the appropriate validation message when the `Day`, `Month`, and `Year` fields are blurred and have values greater than the valid limits", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "", mm: "", yyyy: "" }}
+          onChange={() => {}}
+          enableInternalWarning
+        />
+      </CarbonProvider>
+    );
+
+    const dayInput = screen.getByRole("textbox", { name: "Day" });
+    const monthInput = screen.getByRole("textbox", { name: "Month" });
+    const yearInput = screen.getByRole("textbox", { name: "Year" });
+    await user.type(dayInput, "32");
+    await user.type(monthInput, "13");
+    await user.type(yearInput, "2201");
+    await user.tab();
+
+    expect(
+      screen.getByText(
+        "Day should be a number within a 1-31 range. " +
+          "Month should be a number within a 1-12 range. " +
+          "Year should be a number within a 1800-2200 range."
+      )
+    ).toBeVisible();
+  });
+});
+
+test("should submit the form when enter key is pressed", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  const onSubmit = jest.fn();
```

Comment:
```suggestion
  const onSubmit = jest.fn((e) => { e.preventDefault(); });
```

**suggestion** (non-blocking): it probably doesn't matter is the test is obviously passing and I have no idea how jsdom handles form submission, but I'd feel a little more comfortably if we were explicitly preventing default so that we know there's definitely going to be no attempt at reloading the page etc.

Reply:
As it's running in a node environment, there's browser behaviour so there's nothing to prevent when a form submission occurs